src/queue.c

```diff
@@ -6216,10 +6247,61 @@ _dispatch_worker_thread(void *context)
 	_dispatch_sigmask();
 #endif
 	_dispatch_introspection_thread_add();
+	dispatch_priority_t pri = dq->dq_priority;
+	pthread_priority_t pp = _dispatch_get_priority();
+
+	// The Linux kernel does not have a direct analogue to the QoS-based
```

Comment:
Minor nit: since the comment refers to Linux, the following preprocessor directive should come above this line.